### PR TITLE
PROD-851PROD-851 - Fix topic tag creation for non-admin organizer

### LINKED ISSUE #2185: The non-admin group organizers cannot add discussion tags
**To Reproduce**
1. Create a group with a forum.
2. Make a non-admin as the organizer of the group.
3. Switch to the organizer's account
4. Go to the frontend and try to create a new discussion
5. Notice the error

**Screenshots**

**Support ticket links**
https://secure.helpscout.net/conversation/1437843434/127695

### PR DESCRIPTION
**Pull request by ngolwalkar**
### Changes proposed in this Pull Request:

Fixes #2185

### How to test the changes in this Pull Request:

1. Create a group with a forum.
2. Make a non-admin as the organizer of the group.
3. Switch to the organizer's account
4. Go to the frontend and try to create a new discussion
5. You as non-admin organizer now can create topic tags

### Proof Screenshots or Video

https://www.loom.com/share/965c57366a564d37b3c6efe42f3c0296

### CODE REVIEW

**Review by chetansatasiya — COMMENTED** [review]
@ngolwalkar I am curious to know why we need this filter and you said like it's already they have the capablitlity via WP Role then why we are adding this filter.

**Comment by ngolwalkar** [comment]
> I am curious to know why we need this filter and you said like it's already they have the capablitlity via WP Role then why we are adding this filter?

I checked with the capabilities of the current user role, the capability was there on the role.  but the function current_user_can() returns false. so for that, we could check with the map_meta_cap for that specific capability, and the tag area was there with that fix. But again when we add tags, those tags were not getting created in admin, since wp_insert_post again check for creating the terms with the same function current_user_can(), so it was again getting failed to create tags there. As you said there should be a filter that was changing the value for it, I also checked with it but there was no filter used for that. So that is why I came up with assigning the non-admin organizer to assign the moderator role and used this filter.

File: src/bp-forums/core/capabilities.php [hunk 1]
```diff
@@ -608,3 +608,17 @@ function bbp_restore_caps_from_wp_roles() {
 		'message' => __( 'Complete!', 'buddyboss' ),
 	);
 }
+
+// Restore capabilities to create topic tags for non-admin organizer
```
Comment (chetansatasiya) [inline src/bp-forums/core/capabilities.php:612]: @ngolwalkar PLease follow the proper function doc block with since tag and with all the necessary info.

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need changes

**Review by chetansatasiya — APPROVED** [review]

**Pull request closed without merging**